### ISSUE TITLE
make error compatible with `hardhat-chai-machers`

### ISSUE THREAD

**Issue by shunjizhan** [AcalaNetwork/bodhi.js#743]
so it's more compatible with hardhat out of box, otherwise we will need to manually import `'@nomiclabs/hardhat-waffle'` and install extra pkgs;

**Comment by ermalkaleci** [comment]
it needs to be similar to Ethereum

**Comment by shunjizhan** [comment]
Yeah I think our error format is already same as eth
```
{
    "jsonrpc": "2.0",
    "id": 1,
    "error": {
        "code": -32000,
        "message": "some error msg"
    }
}
```
but hardhat matchers looks for `error.data` instead, not sure how it works for eth, need more digging

https://github.com/NomicFoundation/hardhat/blob/e92b6b7b90032e380d43919b11a0d7a61f093ff6/packages/hardhat-chai-matchers/src/internal/reverted/utils.ts#L29

**Comment by shunjizhan** [comment]
oh actually eth do have a `error.data` field

**Comment by shunjizhan** [comment]
closed by #756
